### ISSUE TITLE
SSD training with images of arbitrary size?

### ISSUE THREAD

**Issue by 1292765944** [weiliu89/caffe#409]
Why not you train SSD with arbitrary size? Is there any hard tricks? I think it should achieve higher mAP in pascal voc. Thank you very much!

**Comment by weiliu89** [comment]
Are you suggesting doing some data augmentation as what Faster RCNN or YOLOv2 are doing? Randomly resize the image to different sizes during training? I think that it is worthwhile to try, so that the net can response well to objects of different sizes. Although I think current augmentation does something similar to this.

**Comment by ksaluja15** [comment]
Don't you think your new expansion trick and batch sampling essentially mimics training with arbitrary sizes ? @weiliu89

**Comment by weiliu89** [comment]
@ksaluja15 Yes. I think they are very similar (as what I said above), although not exactly the same.

**Comment by 1292765944** [comment]
@weiliu89 Yes, I agree that the sampling strategy (random expansion) explores the arbitrary size and aspect ratios. But I think when in testing, current models must resize the images to fixed size which will warp and loss the original information. I do not think the current sampling can cover all the situations. While keep the aspect ratio is a more general approach.

**Comment by weiliu89** [comment]
@1292765944 Yeah. I have a script (ssd_pascal_orig.py) which tries to train SSD with original aspect ratio. But again, it is a tradeoff between accuracy and speed. Resizing images into same size can take advantage of batch processing and thus be faster in both training and inference. Of course, it might also affect the performance a bit.

**Comment by miquelmarti** [comment]
@weiliu89 What about not only keeping the original aspect ratio but also the same size, have you tried that?
